From 8d09e919fca2cb9c9363df831de30beaf0672f68 Mon Sep 17 00:00:00 2001
From: Tanuj Tekriwal <tanuj.tekriwal@intel.com>
Date: Wed, 24 May 2023 12:05:31 +0530
Subject: [PATCH] Removing build for drm_hwcomposer default

Since intel uses its own drm-hwcomposer so this is
not needed. Intel path is vendor/intel/external/drm-hwcomposer

Tracked-On: NA
Signed-off-by: Tanuj Tekriwal <tanuj.tekriwal@intel.com>
---
 Android.bp       | 170 -----------------------------------------------
 tests/Android.bp |  44 ------------
 2 files changed, 214 deletions(-)
 delete mode 100644 Android.bp
 delete mode 100644 tests/Android.bp

diff --git a/Android.bp b/Android.bp
deleted file mode 100644
index a4d8e18..0000000
--- a/Android.bp
+++ /dev/null
@@ -1,170 +0,0 @@
-// Copyright (C) 2015 The Android Open Source Project
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//      http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
-// =====================
-// libdrmhwc_utils.a
-// =====================
-package {
-    default_applicable_licenses: ["external_drm_hwcomposer_license"],
-}
-
-// Added automatically by a large-scale-change
-// See: http://go/android-license-faq
-license {
-    name: "external_drm_hwcomposer_license",
-    visibility: [":__subpackages__"],
-    license_kinds: [
-        "SPDX-license-identifier-Apache-2.0",
-    ],
-    license_text: [
-        "NOTICE",
-    ],
-}
-
-cc_library_static {
-    name: "libdrmhwc_utils",
-
-    srcs: ["utils/Worker.cpp"],
-
-    include_dirs: [
-        "external/drm_hwcomposer",
-        "external/drm_hwcomposer/include",
-    ],
-
-    cflags: [
-        "-Wall",
-        "-Werror",
-    ],
-
-    vendor: true,
-
-}
-
-// =====================
-// hwcomposer.drm.so
-// =====================
-cc_defaults {
-    name: "hwcomposer.drm_defaults",
-
-    shared_libs: [
-        "libcutils",
-        "libdrm",
-        "libhardware",
-        "libhidlbase",
-        "liblog",
-        "libsync",
-        "libui",
-        "libutils",
-    ],
-
-    include_dirs: [
-        "external/drm_hwcomposer",
-        "external/drm_hwcomposer/include",
-    ],
-
-    static_libs: ["libdrmhwc_utils"],
-
-    cflags: [
-        "-Wall",
-        "-Werror",
-    ],
-
-    cppflags: [
-        "-DHWC2_INCLUDE_STRINGIFICATION",
-        "-DHWC2_USE_CPP11",
-    ],
-
-    product_variables: {
-        platform_sdk_version: {
-            cflags: ["-DPLATFORM_SDK_VERSION=%d"],
-        },
-    },
-
-    relative_install_path: "hw",
-    vendor: true,
-}
-
-cc_library_static {
-    name: "drm_hwcomposer",
-    defaults: ["hwcomposer.drm_defaults"],
-    srcs: [
-        "DrmHwcTwo.cpp",
-
-        "bufferinfo/BufferInfoGetter.cpp",
-        "bufferinfo/BufferInfoMapperMetadata.cpp",
-
-        "compositor/DrmDisplayComposition.cpp",
-        "compositor/DrmDisplayCompositor.cpp",
-        "compositor/Planner.cpp",
-
-        "drm/DrmConnector.cpp",
-        "drm/DrmCrtc.cpp",
-        "drm/DrmDevice.cpp",
-        "drm/DrmEncoder.cpp",
-        "drm/DrmEventListener.cpp",
-        "drm/DrmFbImporter.cpp",
-        "drm/DrmMode.cpp",
-        "drm/DrmPlane.cpp",
-        "drm/DrmProperty.cpp",
-        "drm/ResourceManager.cpp",
-        "drm/VSyncWorker.cpp",
-
-        "utils/autolock.cpp",
-        "utils/hwcutils.cpp",
-
-        "backend/Backend.cpp",
-        "backend/BackendClient.cpp",
-        "backend/BackendManager.cpp",
-        "backend/BackendRCarDu.cpp",
-    ],
-}
-
-cc_library_shared {
-    name: "hwcomposer.drm",
-    defaults: ["hwcomposer.drm_defaults"],
-    whole_static_libs: ["drm_hwcomposer"],
-    srcs: ["bufferinfo/legacy/BufferInfoLibdrm.cpp"],
-}
-
-cc_library_shared {
-    name: "hwcomposer.drm_minigbm",
-    defaults: ["hwcomposer.drm_defaults"],
-    whole_static_libs: ["drm_hwcomposer"],
-    srcs: ["bufferinfo/legacy/BufferInfoMinigbm.cpp"],
-    include_dirs: ["external/minigbm/cros_gralloc"],
-}
-
-// Used by hwcomposer.drm_imagination
-filegroup {
-    name: "drm_hwcomposer_platformimagination",
-    srcs: ["bufferinfo/legacy/BufferInfoImagination.cpp"],
-}
-
-// Used by hwcomposer.drm_hikey and hwcomposer.drm_hikey960
-filegroup {
-    name: "drm_hwcomposer_platformhisi",
-    srcs: ["bufferinfo/legacy/BufferInfoMaliHisi.cpp"],
-}
-
-// Used by hwcomposer.drm_meson
-filegroup {
-    name: "drm_hwcomposer_platformmeson",
-    srcs: ["bufferinfo/legacy/BufferInfoMaliMeson.cpp"],
-}
-
-// Used by hwcomposer.drm_mediatek
-filegroup {
-    name: "drm_hwcomposer_platformmediatek",
-    srcs: ["bufferinfo/legacy/BufferInfoMaliMediatek.cpp"],
-}
diff --git a/tests/Android.bp b/tests/Android.bp
deleted file mode 100644
index 473c944..0000000
--- a/tests/Android.bp
+++ /dev/null
@@ -1,44 +0,0 @@
-cc_library_shared {
-    name: "hwcomposer.filegroups_build_test",
-    defaults: ["hwcomposer.drm_defaults"],
-    whole_static_libs: ["drm_hwcomposer"],
-
-    srcs: [
-        ":drm_hwcomposer_platformhisi",
-        ":drm_hwcomposer_platformimagination",
-        ":drm_hwcomposer_platformmediatek",
-        ":drm_hwcomposer_platformmeson",
-    ],
-
-    local_include_dirs: [
-        "test_include",
-    ],
-
-    cppflags: [
-        "-DDISABLE_LEGACY_GETTERS",
-    ],
-}
-
-package {
-    // See: http://go/android-license-faq
-    // A large-scale-change added 'default_applicable_licenses' to import
-    // all of the 'license_kinds' from "external_drm_hwcomposer_license"
-    // to get the below license kinds:
-    //   SPDX-license-identifier-Apache-2.0
-    default_applicable_licenses: ["external_drm_hwcomposer_license"],
-}
-
-cc_test {
-    name: "hwc-drm-tests",
-
-    srcs: ["worker_test.cpp"],
-
-    vendor: true,
-    header_libs: ["libhardware_headers"],
-    static_libs: ["libdrmhwc_utils"],
-    shared_libs: ["hwcomposer.drm"],
-    include_dirs: [
-        "external/drm_hwcomposer",
-        "external/drm_hwcomposer/include",
-    ],
-}
-- 
2.17.1

